tests/unit/challenges/test_models.py

```diff
@@ -0,0 +1,142 @@
+import os
+import shutil
+
+from datetime import timedelta
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.test import TestCase
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+from challenges.models import Challenge, ChallengePhase
+from hosts.models import ChallengeHost, ChallengeHostTeam
+
+
+class BaseTestCase(TestCase):
+
+    def setUp(self):
+
+        self.user = User.objects.create(
+            username='someuser',
+            email="user@test.com",
+            password='secret_password')
+
+        self.challenge_host_team = ChallengeHostTeam.objects.create(
+            team_name='Test Challenge Host Team',
+            created_by=self.user)
+
+        self.challenge = Challenge.objects.create(
+            title='Test Challenge',
+            description='Description for test challenge',
+            terms_and_conditions='Terms and conditions for test challenge',
+            submission_guidelines='Submission guidelines for test challenge',
+            creator=self.challenge_host_team,
+            published=False,
+            enable_forum=True,
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+            anonymous_leaderboard=False)
+
+
+class ChalengeTestCase(BaseTestCase):
+
+    def setUp(self):
+        super(ChalengeTestCase, self).setUp()
+
+        try:
+            os.makedirs('/tmp/evalai')
+        except OSError:
+            pass
+
+        with self.settings(MEDIA_ROOT='/tmp/evalai'):
+            self.challenge.image = SimpleUploadedFile('test_sample_file.jpg',
```

Comment:
Probably more rigorous tests here would be nice! For example: 
Can we check if random characters return an error or not? 
Can we check if the length of the URL is okay? 

We should probably put a check for folder name being not too long! and image name being not too long!